courses-fe/src/pages/login/LoginComponent.js

```diff
@@ -1,73 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import './style.scss';
 
-class LoginPage extends React.Component {
-    constructor(props) {
-        super(props);
+const LoginPage = (props) => {
+    const [userName, setUserName] = useState('');
+    const [password, setPassword] = useState('');
+    const [submitted, setSubmitted] = useState(false);
 
-        // reset login status
-
-        this.state = {
-            username: '',
-            password: '',
-            submitted: false
-        };
-
-        this.handleChange = this.handleChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
-
-    handleChange(e) {
+    const handleChange = (e) => {
         const { name, value } = e.target;
-        this.setState({ [name]: value });
+        setUserName({ [name]: value });
     }
 
-    handleSubmit(e) {
+    const handleSubmit = (e) => {
         e.preventDefault();
-
-        this.setState({ submitted: true });
-        const { username, password } = this.state;
-        if (username && password) {
-            this.props.login(username, password);
+        setSubmitted(true);
+        //const { username, password } = this.state;
+        if (userName && password) {
+            this.props.login(userName, password);
```

Comment:
nu mai trebuie folosit this in componente de tip hooks

Reply:
de asemenee aici functia asta de props.login o sa iti dea eroare ca nu o ai inca in redux ... nu ai pus-i in mapDispatchToProps deci ea nu exista p props....ar trebui scoasa deocaamdataa pana creezi api-uurile si redux thunk